### PR TITLE
Wild pokemon area

### PR DESCRIPTION
**Pull request by PikalaxALT**

### CODE REVIEW

File: src/wild_pokemon_area.c [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+#include "roamer.h"
+#include "overworld.h"
+#include "pokedex_area_markers.h"
+#include "constants/vars.h"
+#include "constants/region_map.h"
+#include "constants/species.h"
+
+struct SeviiDexArea
+{
+    const u16 (* lut)[2];
```
Comment (jiangzhengwenjz) [inline src/wild_pokemon_area.c]: use of space is not consistent with line 29 and 295

File: src/wild_pokemon_area.c [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+        }
+    }
+
+    return areaCount;
+}
+
+static s32 GetRoamerIndex(u16 species)
+{
+    s32 i;
+    for (i = 0; i < 3u; i++)
```
Comment (jiangzhengwenjz) [inline src/wild_pokemon_area.c]: NELEMS

File: src/wild_pokemon_area.c [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+    u16 roamer;
+    u16 starter;
+};
+
+static s32 GetRoamerIndex(u16 species);
+static s32 CountRoamerNests(u16 species, struct Subsprite * subsprites);
+static bool32 PokemonInAnyEncounterTableInMap(const struct WildPokemonHeader * data, s32 species);
+static bool32 PokemonInEncounterTable(const struct WildPokemonInfo * pokemon, s32 species, s32 count);
+static u16 GetMapSecIdFromWildMonHeader(const struct WildPokemonHeader * header);
+static bool32 TryGetMapSecPokedexAreaEntry(u16 mapSecId, const u16 (*lut)[2], s32 count, s32 * lut_i_p, u16 * table_idx_p);
```
Comment (jiangzhengwenjz) [inline src/wild_pokemon_area.c]: Shouldn't formal parameters be in camelCase?

Comment (PikalaxALT) [reply src/wild_pokemon_area.c]: For these the `_p` is kinda important as it communicates that this points to a variable that serves as `lut_i` or `table_idx` in both contexts. So to follow your convention, I would need `lutI_p` and `tableIdx_p`. Is that suitable?

**Pull request merged**